### ISSUE TITLE
java.lang.ClassCastException: class tec.units.indriya.unit.ProductUnit cannot be cast to class tec.uom.se.AbstractUnit (tec.units.indriya.unit.ProductUnit and tec.uom.se.AbstractUnit are in unnamed module of loader 'app')

### ISSUE THREAD

**Issue by Azbesciak** [unitsofmeasurement/indriya#158]
Guys, tell me what I suppose to do- I have multiproject build in gradle,In one I am using [geotools 20.0](https://mvnrepository.com/artifact/org.geotools/gt-shapefile/20.0) which uses :
```
si.uom:si-quantity:0.7.1
uom:si-units-java8:0.7.1
systems.uom:systems-common-java8:0.7.2
tec.uom.lib:uom-lib-common:1.0.2
tec.uom:uom-se:1.0.8
```
In another, I am using [Physikal SI Units kotlin extension](compile group: 'org.tenkiv.physikal', name: 'si-units', version: '2.0.2.0'
) which uses:
```
si.uom:si-quantity:0.9
si.uom:si-units:0.9
tec.units:indriya:1.0
tec.uom.lib:uom-lib-common:1.0.2
```

For last 5 hours I struggle with following error during classes initialization:
`class tec.units.indriya.unit.ProductUnit cannot be cast to class tec.uom.se.AbstractUnit (tec.units.indriya.unit.ProductUnit and tec.uom.se.AbstractUnit are in unnamed module of loader 'app')`

I have tried to manualy manage my dependencies, finished at 
```
implementation(group: 'org.geotools', name: 'gt-shapefile', version: '20.0') {
        exclude group: 'si.uom'
        exclude group: 'tec.uom'
}
compile (group: 'si.uom', name: 'si-units', version: '0.9') // higher version, but only changes 
compile (group: 'tec.uom', name: 'uom-se', version: '1.0.8')
```
... and no Ideas what to do farther.
So, tell me, how should I manage those self discovery services etc?
I am using Java 11 and gradle 4.10.2, but all modules are anonymous.

**Comment by Azbesciak** [comment]
Finished with 'compile systems.uom:systems-common-java8:0.8'` in shared module, 
```
implementation(group: 'org.tenkiv.physikal', name: 'si-units', version: '2.0.2.0') {
	exclude group: 'si.uom'
	exclude group: 'tec.uom'
	exclude group: 'tec.uom.lib'
}
```
and
```
implementation(group: 'org.geotools', name: 'gt-shapefile', version: '20.0') {
       exclude group: 'systems.uom'
 }
```
But anyway would want to know the answer as long as this spaghetti didn't blow up.

**Comment by desruisseaux** [comment]
I have not investigated above issue, but just for information: if you are using GeoTools for coordinate transformations, [Apache SIS](http://sis.apache.org/) is a better alternative and is bundled with its own JSR-363 implementation (should not conflict with other implementations if you have many on the classpath). If you are using GeoTools for reading ShapeFiles, then unfortunately SIS is not yet ready for that.

**Comment by Azbesciak** [comment]
I upgraded geotools to 20.0 yesterday, and unfortunatly, they are using 0.7.1/0.7.2 (all of mentioned above) ... I tried manualy manage 0.9, but with mentioned earlier result.

**Comment by keilw** [comment]
Then it sounds like this is up to GeoTools to fix it with a 20.1 or similar upgrade. @desruisseaux Any chance to help with that? The only temporary fix I could think of is changing the version of Physikal to something between 1.0 and 1.1, see https://search.maven.org/search?q=g:org.tenkiv.physikal%20AND%20a:core&core=gav that uses the same SI and Unit Systems versions and is still based on uom-se.
